### ISSUE TITLE
How to convert multiple models to a single model ?

### ISSUE THREAD

**Issue by onlie202** [MIC-DKFZ/nnUNet#299]
Every fold training produce a model, totally 5. How can I convert them to a single one to save time for prediction ? ( I'm using 3D fullres. )

**Comment by FabianIsensee** [comment]
Hi,
nnU-Net never converts them to a single model and instead uses them as an ensemble of 5 models. If you need a single model, please specify 'all' as fold. This will train on all training data. Note that validation scores cannot be used from this training as it will validate on its own training cases!
Best,
Fabian

**Comment by onlie202** [comment]
Thanks for your reply !

**Comment by rohan19250** [comment]
Hi Fabian,

Somewhat related to this, I had a couple of clarifications. Once I have 3d full res configuration model for each of the 5 folds? Is it possible to get/calculate an average final dice score of these 5 folds, since dice scores from each fold seem to be quite different?

Secondly, the best way to use an ensemble of these 5 folds is to:
(https://github.com/MIC-DKFZ/nnUNet#run-inference)
- nnUNet_predict
- nnUNet_ensemble ?
But is this only if we want to do predictions on a test set?

Lastly, what files do -pp option (Post-processing) generate?

**Comment by FabianIsensee** [comment]
when you run nnUNet_determine_postprocessing for this model and task it will be done for you. There will be a folder cv_niftis_raw along with the fold_X folders. In there is a summary.json file with the information you need

Ensemble of 5 folds only needs nnUNet_predict. nnUNet_ensemble is only to ensemble different configurations (lowres, fullres, 2d, cascade). When running nnUNet_predict, you can select the folds it should use with -f 0 1 2 3 4

**Comment by FabianIsensee** [comment]
-pp needs an input file, not an output file. It's the postprocessing.json one. This is only relevant for nnUNet_ensemble. You get that by letting nnunet automatically determine what ensemble to use (forgot the command name right now)
